compiler/rustc_resolve/src/imports.rs

```diff
@@ -313,16 +313,14 @@ impl<'a, 'tcx> Resolver<'a, 'tcx> {
                     (true, true) => {
                         // FIXME: remove `!binding.is_ambiguity()` after delete the warning ambiguity.
                         if !binding.is_ambiguity()
+                            && !old_binding.determined()
```

Comment:
Is this change orthogonal to the change in `ident.rs`?
What does it mean?
The condition on this branch feels increasingly more random.

Reply:
How would this code look like if we didn't care about backward compatibility?

Reply:
> Is this change orthogonal to the change in ident.rs?

I don't think it's orthogonal, and removing `!old_binding.determined()` all the tests run through it as well. It's only written this way to be a bit stricter, as I'm not sure if there are any boundary cases.

> The condition on this branch feels increasingly more random.

I also want to remove this branch, but I haven't found a better way than overwriting the binding...

> How would this code look like if we didn't care about backward compatibility?

I think can only remove `binding.is_ambiguity()`.